tools/functional-tester/etcd-runner/main.go

```diff
@@ -310,10 +320,24 @@ func runWatcher(eps []string) {
 						return
 					}
 				}
-			}(prefixes[i])
+			}(roundPrefix + "-" + prefixes[i])
 		}
 	}
 	wg.Wait()
+
+	// cancel all watch channels
+	cancelc()
+
```

Comment:
after we cancel the ctx at line 329, all the watch chans should be closed.

so at line 330 we should do

```
// wcs contains all watch channels created at line 307

for _, wc := wcs {
    select {
        case e, ok <- := wc:
            if !ok {
                log.Fatalf("expected wc to be closed, but received %v, e)
            }
        case <-time.After(10*time.Second):
             log.Fatal("timed out waiting for watcher cancellation")
    }
}

```